atc/engine/engine.go

```diff
@@ -238,7 +238,7 @@ func (b *engineBuild) Run(ctx context.Context) {
 		logger.Info("releasing")
 
 	case <-done:
-		if errors.As(runErr, &exec.Retriable{}) {
+		if errors.As(runErr, &exec.Retriable{}) && !b.build.IsAborted() {
```

Comment:
Makes sense, but I think it might be cleaner to push this logic down into the `RetryErrorStep` where it can check if `ctx.Err() != nil` to avoid wrapping it in a `Retriable` in the first place.

The `RetryStep` does something similar:

https://github.com/concourse/concourse/blob/0e8560d9b2ca285870a752a43fa0954642ac7297/atc/exec/retry_step.go#L30-L32

Reply:
Good point. Just updated the code change.